yast/src/modules/SMTData.rb

```diff
@@ -0,0 +1,1480 @@
+# encoding: utf-8
+
+# File:	modules/SMTData.ycp
+# Package:	Configuration of SMT
+# Summary:	SMT global data and functions
+# Authors:	Lukas Ocilka <locilka@suse.cz>
+#
+# $Id:$
+require "yast"
+
+module Yast
+  class SMTDataClass < Module
+    def main
+      Yast.import "UI"
+      textdomain "smt"
+
+      Yast.import "Message"
+      Yast.import "Service"
+      Yast.import "Report"
+      Yast.import "FileUtils"
+      Yast.import "String"
+      Yast.import "Label"
+      Yast.import "Popup"
+      Yast.import "PackageSystem"
+      Yast.import "Wizard"
+      Yast.import "Progress"
+      Yast.import "Directory"
+      Yast.import "ProductControl"
+
+      @all_credentials = {}
+
+      @smt_conf = "/etc/smt.conf"
+
+      # script entered into a crontab
+      @cron_mirror_script = "/usr/bin/smt-mirror.pl"
+      # cron
+      @path_to_cron = "/etc/smt.d/novell.com-smt"
+
+      @path_to_ncc_sync_script = "/usr/sbin/smt-ncc-sync"
+      @path_to_scc_sync_script = "/usr/sbin/smt-scc-sync"
+      @user_for_sync_script = "root"
+
+      @server_cert = "/etc/ssl/certs/YaST-CA.pem"
+      @apache_cert = "/srv/www/htdocs/smt.crt"
+
+      # own control file
+      @smt_control_file = "/usr/share/YaST2/control/smt_control.xml"
+
+      @smt_cron_settings = []
+
+      @first_run = nil
+
+      @first_run_file = Builtins.sformat(
+        "%1/smt-already-adjusted",
+        Directory.vardir
+      )
+
+      @smt_enabled = nil
+
+      @smt_services = ["smt"]
+
+      @smt_database = "mysql"
+
+      @SCCcredentials_file = "/etc/zypp/credentials.d/SCCcredentials"
+      @NCCcredentials_file = "/etc/zypp/credentials.d/NCCcredentials"
+
+      @read_api_type = "NCC"
+
+      @initial_config = false
+
+      @_smt_user = nil
+      @_smt_user_loaded = false
+
+      @smt_enabled_initial = nil
+
+      @ca_already_called = false
+
+      @mysql_root_password = ""
+
+      # bnc #389804
+      @database_already_exists = false
+
+      @initial_password = nil
+    end
+
+    def GetSCCcredentialsFile
+      @SCCcredentials_file
+    end
+
+    def SystemIsRegistered
+      FileUtils.Exists(@SCCcredentials_file) || FileUtils.Exist(@NCCCredentials_file)
+    end
+
+    def InitialConfig
+      @initial_config
+    end
+
+    def ApiTypeChanged
+      return false if InitialConfig()
+      @read_api_type != Ops.get(@all_credentials, ["NU", "ApiType"], "NCC")
```

Comment:
Aha, good to know. I can add that migration will add ApiType to NCC if ApiType is missing during upgrades. 
But at the end I need ApiType to be SCC since NCC related modules are gone from SMT12. (AFAICT, SMT12 can not talk to NCC anymore. Don't know if that's correct, but I understood it as desired behaviour)